test/integration/httproute_test.go

```diff
@@ -203,6 +204,29 @@ func TestHTTPRouteEssentials(t *testing.T) {
 		helpers.EventuallyGETPath(t, proxyURL, "", http.StatusOK, "<title>httpbin.org</title>", map[string]string{"Content-Type": "audio/mp3"}, ingressWait, waitTick)
 	})
 
+	t.Run("HTTPRoute query param match", func(t *testing.T) {
+		RunWhenKongExpressionRouter(t)
+		RunWhenKongVersion(t, fmt.Sprintf(">=%s", versions.QueryParameterVersionCutoff))
```

Comment:
Why not use the already existing `RunWhenKongExpressionRouterWithVersion`?

Reply:
This `RunWhenKongExpressionRouterWithVersion` skips the test only when the router flavor is `expressions` AND Kong version is not in the range. (`router_flavor != "expressions" ||  KongVersionOK`)
Here we need to skip the tests when router flavor is  not `expressions` OR Kong version is not in the range. Seems confusing, could you suggest a better naming for the conditional skipping funcs?

Reply:
As @programmer04 mentioned this will no longer be needed - `RunWhenKongExpressionRouter` will be enough.